clang/lib/Driver/Driver.cpp

```diff
@@ -1593,6 +1593,10 @@ Compilation *Driver::BuildCompilation(ArrayRef<const char *> ArgList) {
     }
   }
 
+  bool IsSYCL =
```

Comment:
nit - This boolean isn't used anywhere else, you could pull the hasFlag check into the if statement